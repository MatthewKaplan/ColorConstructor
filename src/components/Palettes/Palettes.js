import React, { Component } from "react";

class Palettes extends Component {
  render() {
    const {id, hex, isLocked, lockPalette} = this.props;
    return (
      <div className="palettes-container" style={{ backgroundColor: hex }}>
        <h1 className="hex-code">{hex.toUpperCase()}</h1>
        <i role="button" className="Color-lock-icon material-icons" onClick={() => lockPalette(id)}>
				{isLocked ? 'lock' : 'lock_open'}
			</i>
      </div>
    );
  }
}

export default Palettes;
