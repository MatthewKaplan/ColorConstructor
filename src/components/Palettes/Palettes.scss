.palettes-container {
	display: flex;
	justify-content: center;
	align-items: center;
	flex-direction: column;
	padding: 25px;
	color: #ffffff;
	transition-duration: .5s;
	transition-timing-function: ease-in;
	box-shadow: 5px 5px 5px black;
	height: 100%;

	&:first-child {
		border-radius: 8px 0 0 8px;
	}

	&:nth-child(5) {
		border-radius: 0 8px 8px 0;
	}
}

.hex-code {
	cursor: default;
}

.Color-lock-icon {
	font-size: 50px;
	cursor: pointer;
  transition: background-color 0.1s;
  padding: 3px;
  border-radius: 5px;

	&:hover {
    background-color: #b6b6b66b;
	}
}
