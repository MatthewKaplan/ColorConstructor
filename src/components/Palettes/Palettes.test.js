import React from "react";
import { shallow } from "enzyme";
import Palettes from "./Palettes";

const mockHex = "#AD8B72";
const mockLockPalette = jest.fn();

describe("Palettes", () => {
  let wrapper, instance;

  it("should match the snapshot", () => {
    wrapper = shallow(<Palettes hex={mockHex} lockPalette={mockLockPalette} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should match the snapshot if isLocked is true", () => {
    wrapper = shallow(<Palettes hex={mockHex} lockPalette={mockLockPalette} isLocked={true} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should invoke 'lockPalette' when lock-icon is clicked", () => {
    wrapper = shallow(<Palettes hex={mockHex} lockPalette={mockLockPalette} />);
    wrapper.find("[data-test='lock-icon']").simulate("click");
    expect(mockLockPalette).toHaveBeenCalled();
  });
});
