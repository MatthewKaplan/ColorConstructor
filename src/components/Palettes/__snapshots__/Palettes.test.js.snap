// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Palettes should match the snapshot 1`] = `ShallowWrapper {}`;

exports[`Palettes should match the snapshot if isLocked is true 1`] = `ShallowWrapper {}`;
